reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerFormDecoderProvider.java

```diff
@@ -0,0 +1,403 @@
+		 *
+		 * @param maxSize the maximum in-memory size
+		 * @return {@code this}
+		 */
+		Builder maxSize(long maxSize);
+
+		/**
+		 * Sets the scheduler to be used for offloading disk operations in the decoding phase.
+		 * Default to {@link Schedulers#boundedElastic()}
+		 *
+		 * @param scheduler the scheduler to be used for offloading disk operations in the decoding phase
+		 * @return {@code this}
+		 */
+		Builder scheduler(Scheduler scheduler);
+
+		/**
+		 * When set to {@code true}, the data is streamed directly from the parsed input buffer stream,
+		 * which means it is not stored either in memory or file.
+		 * When {@code false}, parts are backed by in-memory and/or file storage. Default to {@code false}.
+		 * <p><strong>NOTE</strong> that with streaming enabled, the provided {@link Attribute}/{@link FileUpload}
+		 * might not be in a complete state i.e. {@link HttpData#isCompleted()} has to be checked.
+		 * <p>Also note that enabling this property effectively ignores
+		 * {@link #maxInMemorySize(long)},
+		 * {@link #baseDirectory(Path)}, and
+		 * {@link #scheduler(Scheduler)}.
+		 */
+		Builder streaming(boolean enable);
+	}
+
+	final Path baseDirectory;
+	final Charset charset;
+	final long maxInMemorySize;
+	final long maxSize;
+	final Scheduler scheduler;
+	final boolean streaming;
+
+	HttpServerFormDecoderProvider(Build build) {
+		this.baseDirectory = build.baseDirectory;
+		this.charset = build.charset;
+		this.maxInMemorySize = !build.streaming ? build.maxInMemorySize : -1;
+		this.maxSize = build.maxSize;
+		this.scheduler = build.scheduler;
+		this.streaming = build.streaming;
+	}
+
+	/**
+	 * Returns the configured directory where to store disk {@link Attribute}/{@link FileUpload}.
+	 *
+	 * @return the configured directory where to store disk {@link Attribute}/{@link FileUpload}
+	 */
+	@Nullable
+	public Path baseDirectory() {
+		return baseDirectory;
+	}
+
+	/**
+	 * Returns the configured charset for {@link Attribute}/{@link FileUpload}.
+	 *
+	 * @return the configured charset for {@link Attribute}/{@link FileUpload}
+	 */
+	public Charset charset() {
+		return charset;
+	}
+
+	/**
+	 * Returns the configured maximum in-memory size per {@link Attribute}/{@link FileUpload}.
+	 *
+	 * @return the configured maximum in-memory size per {@link Attribute}/{@link FileUpload}
+	 */
+	public long maxInMemorySize() {
+		return maxInMemorySize;
+	}
+
+	/**
+	 * Returns the configured maximum size per {@link Attribute}/{@link FileUpload}.
+	 *
+	 * @return the configured maximum size per {@link Attribute}/{@link FileUpload}
+	 */
+	public long maxSize() {
+		return maxSize;
+	}
+
+	/**
+	 * Returns the configured scheduler to be used for offloading disk operations in the decoding phase.
+	 *
+	 * @return the configured scheduler to be used for offloading disk operations in the decoding phase
+	 */
+	public Scheduler scheduler() {
+		return scheduler;
+	}
+
+	/**
+	 * Returns whether the streaming mode is enabled.
+	 *
```

Comment:
maybe reiterate here what streaming mode == true means?

Reply:
When `streaming==true` Reactor Netty emits parts that might not be complete i.e. the current buffer is parsed and immediately emitted while when `streaming==false` the part will be emitted only when it is in complete form meanwhile all data is stored either in memory or on disk

Reply:
I meant as a javafoc comment 😜

Reply:
aaaa, we can copy from the setter wdyt?

Reply:
for getter methods I'm less descriptive in comparison with setter methods

Reply:
yeah we can copy, or even refer to the setter with a `@see` tag (which doesn't take a description, unlike `{@link ...}`)

Reply:
@simonbasle this should be OK now with the latest commit